mux/src/tmux_commands.rs

```diff
@@ -0,0 +1,304 @@
+                .next()
+                .ok_or_else(|| anyhow!("missing pane_index"))?
+                .parse()?;
+            let cursor_x = fields
+                .next()
+                .ok_or_else(|| anyhow!("missing cursor_x"))?
+                .parse()?;
+            let cursor_y = fields
+                .next()
+                .ok_or_else(|| anyhow!("missing cursor_y"))?
+                .parse()?;
+            let pane_width = fields
+                .next()
+                .ok_or_else(|| anyhow!("missing pane_width"))?
+                .parse()?;
+            let pane_height = fields
+                .next()
+                .ok_or_else(|| anyhow!("missing pane_height"))?
+                .parse()?;
+            let pane_left = fields
+                .next()
+                .ok_or_else(|| anyhow!("missing pane_left"))?
+                .parse()?;
+            let pane_top = fields
+                .next()
+                .ok_or_else(|| anyhow!("missing pane_top"))?
+                .parse()?;
+
+            // These ids all have various sigils such as `$`, `%`, `@`,
+            // so skip those prior to parsing them
+            let session_id = session_id[1..].parse()?;
+            let window_id = window_id[1..].parse()?;
+            let pane_id = pane_id[1..].parse()?;
+
+            items.push(PaneItem {
+                session_id,
+                window_id,
+                pane_id,
+                pane_index,
+                cursor_x,
+                cursor_y,
+                pane_width,
+                pane_height,
+                pane_left,
+                pane_top,
+            });
+        }
+
+        log::info!("panes in domain_id {}: {:?}", domain_id, items);
+        let mux = Mux::get().expect("to be called on main thread");
+        if let Some(domain) = mux.get_domain(domain_id) {
+            if let Some(tmux_domain) = domain.downcast_ref::<TmuxDomain>() {
+                return tmux_domain.inner.sync_pane_state(&items);
+            }
+        }
+        anyhow::bail!("Tmux domain lost");
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct CapturePane(TmuxPaneId);
+impl TmuxCommand for CapturePane {
+    fn get_command(&self) -> String {
+        format!("capturep -p -t {} -e -C\n", self.0)
+    }
+
+    fn process_result(&self, domain_id: DomainId, result: &Guarded) -> anyhow::Result<()> {
+        let mux = Mux::get().expect("to be called on main thread");
+        let domain = match mux.get_domain(domain_id) {
+            Some(d) => d,
+            None => anyhow::bail!("Tmux domain lost"),
+        };
+        let tmux_domain = match domain.downcast_ref::<TmuxDomain>() {
+            Some(t) => t,
+            None => anyhow::bail!("Tmux domain lost"),
+        };
+
+        let pane_map = tmux_domain.inner.remote_panes.borrow();
+        if let Some(pane) = pane_map.get(&self.0) {
+            let lock = pane.lock().unwrap();
+            lock.tx.send(result.output.to_owned()).unwrap();
```

Comment:
there are quite a few `unwrap` calls overall.  In a method that returns `Result` we should be able to use either `?` or `.ok_or_else(|| anyhow!("reason"))?` to avoid taking down the whole program.

In cases where the unwrap represents an impossible situation, please replace the unwrap with an `.expect("reason")`.

That will help with reviewing the code, because it will make your assumptions/invariants/intent clearer by capturing them in the code!